use java.security.SecureRandom to be sure

eighthave opened the pull request:
OTR's `r` value should probably be a good random number.

This also switches instantiation of SecureRandom to when it is first used, rather than when the class is instantiated since SecureRandom is not always required.

cobratbq commented:
Could you skip the class InstanceTag for now?

I'd like to propose an additional constructor which allows the caller to provide a random double. It doesn't make sense to persist a one time SecureRandom instance, anyways. I'd rather construct one in session (or somewhere else) and reuse it for each new InstanceTag.

eighthave committed:
use java.security.SecureRandom to be sure

OTR's r value should probably be a good random number :-|

This also switches instantiation of SecureRandom to when it is first used,
rather than when the class is instantiated since SecureRandom is not always
required.

eighthave commented:
Done.
